src/plugins/expressions/common/execution/types.ts

```diff
@@ -17,16 +17,18 @@
  * under the License.
  */
 
-import { ExpressionType } from '../expression_types';
+import { ExpressionType, SerializableState } from '../expression_types';
 import { Adapters, DataAdapter, RequestAdapter } from '../../../inspector/common';
-import { TimeRange, Query, Filter } from '../../../data/common';
 import { SavedObject, SavedObjectAttributes } from '../../../../core/public';
 
 /**
  * `ExecutionContext` is an object available to all functions during a single execution;
  * it provides various methods to perform side-effects.
  */
-export interface ExecutionContext<InspectorAdapters extends Adapters = Adapters> {
+export interface ExecutionContext<
+  InspectorAdapters extends Adapters = Adapters,
+  ExecutionContextSearch extends SerializableState = SerializableState
```

Comment:
I still called this `ExecutionContextSearch`, but at this point `expressions` only knows that it's something serializable. This means less type safety out of the box (unless you remember to pass the generic type param), however it helps us get rid of a circular type dependency and makes for clearer separation of concerns between the plugins.